lib/lutaml/xmi/parsers/xml.rb

```diff
@@ -55,53 +61,52 @@ def serialize_model_packages(model)
           end
         end
 
-        def serialize_model_classes(model)
-          model.xpath('./packagedElement[@xmi:type="uml:Class" or @xmi:type="uml:AssociationClass"]').map do |klass|
-            {
-              xmi_id: klass["xmi:id"],
-              xmi_uuid: klass["xmi:uuid"],
-              name: klass["name"],
-              package: model,
-              attributes: serialize_class_attributes(klass),
-              associations: serialize_model_associations(klass),
-              operations: serialize_class_operations(klass),
-              constraints: serialize_class_constraints(klass),
-              is_abstract: doc_node_attribute_value(klass, "isAbstract"),
-              definition: doc_node_attribute_value(klass, "documentation"),
-              stereotype: doc_node_attribute_value(klass, "stereotype")
-            }
+        # @note xpath ./packagedElement[@xmi:type="uml:Class" or
+        #                               @xmi:type="uml:AssociationClass"]
+        def serialize_model_classes(package, model)
+          package.packaged_element.select { |e|
+            e.type == "uml:Class" || e.type == "uml:AssociationClass"
```

Comment:
Can we do “is_a?(Uml::Class)” something like that?

Reply:
Will implement `is_type?` in `Shale::Mapper` class to handle these issues.

Reply:
We don’t want new methods? Aren’t AssociatedClass etc separate classes?